diff --git a/src/createrepo_c.c b/src/createrepo_c.c
index 2faeb00..61aa82b 100644
--- a/src/createrepo_c.c
+++ b/src/createrepo_c.c
@@ -827,9 +827,9 @@ main(int argc, char **argv)
     const char *xml_compression_suffix = NULL;
     const char *sqlite_compression_suffix = NULL;
     const char *compression_suffix = NULL;
-    cr_CompressionType xml_compression = CR_CW_ZSTD_COMPRESSION;
-    cr_CompressionType sqlite_compression = CR_CW_BZ2_COMPRESSION;
-    cr_CompressionType compression = CR_CW_ZSTD_COMPRESSION;
+    cr_CompressionType xml_compression = CR_CW_NO_COMPRESSION;
+    cr_CompressionType sqlite_compression = CR_CW_NO_COMPRESSION;
+    cr_CompressionType compression = CR_CW_NO_COMPRESSION;
 
     if (cmd_options->compatibility) {
         xml_compression    = CR_CW_GZ_COMPRESSION;
diff --git a/src/parsehdr.c b/src/parsehdr.c
index 2b75e99..180e4ae 100644
--- a/src/parsehdr.c
+++ b/src/parsehdr.c
@@ -666,7 +666,7 @@ cr_package_from_header(Header hdr,
 
     if (hdrrflags & CR_HDRR_LOADHDRID)
         pkg->hdrid = cr_safe_string_chunk_insert(pkg->chunk,
-                                                 headerGetString(hdr, RPMTAG_HDRID));
+                                                 headerGetString(hdr, RPMTAG_SHA1HEADER));
 
     if (hdrrflags & CR_HDRR_LOADSIGNATURES) {
         rpmtd gpgtd = rpmtdNew();
